NPE when a schema isn't found in a schema registry

<!--
  - Please search to see if an issue already exists for the issue you encountered. Duplicate issues will be closed. [Current GitHub Issues](https://github.com/aklivity/zilla/issues)
  - Please make sure your Zilla version is up to date. [Latest Zilla Release](https://github.com/aklivity/zilla/releases/latest)
-->

**Describe the bug**

```
zilla-demo.schema-registry [31/Jul/2024:18:02:20 +0000] CATALOG_KARAPACE_SCHEMA_REGISTRY_UNRETRIEVABLE_SCHEMA_SUBJECT_VERSION Unable to fetch schema for subject core-paymenthub-simple-pay-ipn-request-value with version latest.
org.agrona.concurrent.AgentTerminationException: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.model.avro.internal.avro.Schema.getType()" because "schema" is null
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.doWork(EngineWorker.java:841)
    at org.agrona.core/org.agrona.concurrent.AgentRunner.doWork(AgentRunner.java:304)
    at org.agrona.core/org.agrona.concurrent.AgentRunner.workLoop(AgentRunner.java:296)
    at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:162)
    at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.model.avro.internal.avro.Schema.getType()" because "schema" is null
    at io.aklivity.zilla.runtime.model.avro@0.9.89/io.aklivity.zilla.runtime.model.avro.internal.AvroWriteConverterHandler.serializeJsonRecord(AvroWriteConverterHandler.java:90)
    at io.aklivity.zilla.runtime.catalog.schema.registry@0.9.89/io.aklivity.zilla.runtime.catalog.schema.registry.internal.handler.SchemaRegistryCatalogHandler.encode(SchemaRegistryCatalogHandler.java:318)
    at io.aklivity.zilla.runtime.model.avro@0.9.89/io.aklivity.zilla.runtime.model.avro.internal.AvroWriteConverterHandler.convert(AvroWriteConverterHandler.java:67)
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.cache.KafkaCachePartition.writeProduceEntryContinue(KafkaCachePartition.java:869)
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheClientProduceFactory$KafkaCacheClientProduceFan.onClientInitialData(KafkaCacheClientProduceFactory.java:733)
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheClientProduceFactory$KafkaCacheClientProduceStream.onClientInitialData(KafkaCacheClientProduceFactory.java:1354)
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheClientProduceFactory$KafkaCacheClientProduceStream.onClientMessage(KafkaCacheClientProduceFactory.java:1276)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.handleReadInitial(EngineWorker.java:1260)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.handleRead(EngineWorker.java:1227)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:229)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.doWork(EngineWorker.java:835)
    ... 4 more
    Suppressed: java.lang.Exception: [engine/data#1]        [0x010100000000001b] streams=[consumeAt=0x00003550 (0x0000000000003550), produceAt=0x00003550 (0x0000000000003550)]
            at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.doWork(EngineWorker.java:839)
            ... 4 more
org.agrona.concurrent.AgentTerminationException: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.engine.binding.function.MessageConsumer.accept(int, org.agrona.DirectBuffer, int, int)" because "receiver" is null
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.doWork(EngineWorker.java:841)
    at org.agrona.core/org.agrona.concurrent.AgentRunner.doWork(AgentRunner.java:304)
    at org.agrona.core/org.agrona.concurrent.AgentRunner.workLoop(AgentRunner.java:296)
    at org.agrona.core/org.agrona.concurrent.AgentRunner.run(AgentRunner.java:162)
    at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: java.lang.NullPointerException: Cannot invoke "io.aklivity.zilla.runtime.engine.binding.function.MessageConsumer.accept(int, org.agrona.DirectBuffer, int, int)" because "receiver" is null
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheBootstrapFactory.doEnd(KafkaCacheBootstrapFactory.java:273)
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheBootstrapFactory$KafkaBootstrapMetaStream.doMetaInitialEnd(KafkaCacheBootstrapFactory.java:1176)
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheBootstrapFactory$KafkaBootstrapMetaStream.doMetaInitialEndIfNecessary(KafkaCacheBootstrapFactory.java:1167)
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheBootstrapFactory$KafkaBootstrapStream.onBootstrapInitialEnd(KafkaCacheBootstrapFactory.java:495)
    at io.aklivity.zilla.runtime.binding.kafka@0.9.89/io.aklivity.zilla.runtime.binding.kafka.internal.stream.KafkaCacheBootstrapFactory$KafkaBootstrapStream.onBootstrapInitial(KafkaCacheBootstrapFactory.java:439)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.binding.function.MessageConsumer.lambda$andThen$0(MessageConsumer.java:82)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.handleReadInitial(EngineWorker.java:1263)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.handleRead(EngineWorker.java:1227)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.concurent.ManyToOneRingBuffer.read(ManyToOneRingBuffer.java:229)
    at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.doWork(EngineWorker.java:835)
    ... 4 more
    Suppressed: java.lang.Exception: [engine/data#2]        [0x0102000000000003] streams=[consumeAt=0x0001d078 (0x000000000001d078), produceAt=0x0001d4d8 (0x000000000001d4d8)]
            at io.aklivity.zilla.runtime.engine@0.9.89/io.aklivity.zilla.runtime.engine.internal.registry.EngineWorker.doWork(EngineWorker.java:839)
            ... 4 more
```

**To Reproduce**
Steps to reproduce the behavior:
1. Run the [http.kafka.karapace](https://github.com/aklivity/zilla-examples/tree/main/http.kafka.karapace)
2. POST an object without registering a schema
3. Zilla crashes with NPE

--- a/runtime/model-avro/src/main/java/io/aklivity/zilla/runtime/model/avro/internal/AvroModelHandler.java
+++ b/runtime/model-avro/src/main/java/io/aklivity/zilla/runtime/model/avro/internal/AvroModelHandler.java
@@ -135,26 +135,29 @@ public abstract class AvroModelHandler
         try
         {
             Schema schema = supplySchema(schemaId);
-            switch (schema.getType())
+            if (schema != null)
             {
-            case STRING:
-                status = true;
-                break;
-            case RECORD:
-                GenericRecord record = supplyRecord(schemaId);
-                in.wrap(buffer, index, length);
-                GenericDatumReader<GenericRecord> reader = supplyReader(schemaId);
-                if (reader != null)
+                switch (schema.getType())
                 {
-                    decoderFactory.binaryDecoder(in, decoder);
-                    reader.read(record, decoder);
+                case STRING:
                     status = true;
+                    break;
+                case RECORD:
+                    GenericRecord record = supplyRecord(schemaId);
+                    in.wrap(buffer, index, length);
+                    GenericDatumReader<GenericRecord> reader = supplyReader(schemaId);
+                    if (reader != null)
+                    {
+                        decoderFactory.binaryDecoder(in, decoder);
+                        reader.read(record, decoder);
+                        status = true;
+                    }
+                    progress = index;
+                    extractFields(buffer, index + length, schema);
+                    break;
+                default:
+                    break;
                 }
-                progress = index;
-                extractFields(buffer, index + length, schema);
-                break;
-            default:
-                break;
             }
         }
         catch (IOException | AvroRuntimeException ex)

--- a/runtime/model-avro/src/main/java/io/aklivity/zilla/runtime/model/avro/internal/AvroWriteConverterHandler.java
+++ b/runtime/model-avro/src/main/java/io/aklivity/zilla/runtime/model/avro/internal/AvroWriteConverterHandler.java
@@ -87,34 +87,37 @@ public class AvroWriteConverterHandler extends AvroModelHandler implements Conve
         {
             Schema schema = supplySchema(schemaId);
 
-            switch (schema.getType())
+            if (schema != null)
             {
-            case STRING:
-                next.accept(buffer, index, length);
-                valLength = length;
-                break;
-            case RECORD:
-                GenericDatumReader<GenericRecord> reader = supplyReader(schemaId);
-                GenericDatumWriter<GenericRecord> writer = supplyWriter(schemaId);
-                if (reader != null)
+                switch (schema.getType())
                 {
-                    GenericRecord record = supplyRecord(schemaId);
-                    in.wrap(buffer, index, length);
-                    expandable.wrap(expandable.buffer());
-                    record = reader.read(record, decoderFactory.jsonDecoder(schema, in));
-                    encoderFactory.binaryEncoder(expandable, encoder);
-                    writer.write(record, encoder);
-                    encoder.flush();
-                    int position = expandable.position();
-                    if (position > 0)
+                case STRING:
+                    next.accept(buffer, index, length);
+                    valLength = length;
+                    break;
+                case RECORD:
+                    GenericDatumReader<GenericRecord> reader = supplyReader(schemaId);
+                    GenericDatumWriter<GenericRecord> writer = supplyWriter(schemaId);
+                    if (reader != null)
                     {
-                        next.accept(expandable.buffer(), 0, position);
-                        valLength = position;
+                        GenericRecord record = supplyRecord(schemaId);
+                        in.wrap(buffer, index, length);
+                        expandable.wrap(expandable.buffer());
+                        record = reader.read(record, decoderFactory.jsonDecoder(schema, in));
+                        encoderFactory.binaryEncoder(expandable, encoder);
+                        writer.write(record, encoder);
+                        encoder.flush();
+                        int position = expandable.position();
+                        if (position > 0)
+                        {
+                            next.accept(expandable.buffer(), 0, position);
+                            valLength = position;
+                        }
                     }
+                    break;
+                default:
+                    break;
                 }
-                break;
-            default:
-                break;
             }
         }
         catch (IOException | AvroRuntimeException ex)

